cpp/include/games/stochastic_nim/PerfectPlayer.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <core/AbstractPlayer.hpp>
+#include <core/BasicTypes.hpp>
+#include <games/stochastic_nim/Constants.hpp>
+#include <games/stochastic_nim/Game.hpp>
+#include <util/BitSet.hpp>
+#include <util/EigenUtil.hpp>
+
+namespace stochastic_nim {
+
+class PerfectStrategy {
+ public:
+  struct Params {
+    const int starting_stones;
+    const int max_stones_to_take;
+    const float* chance_event_probs;
+    const int num_chance_events;
+  };
```

Comment:
I don't see the point of having this `Params` class. You have all these readily available in `Constants.hpp`, so why not just use those directly wherever needed?